deployment/src/main/java/io/quarkiverse/logging/splunk/DevServicesLoggingSplunkProcessor.java

```diff
@@ -0,0 +1,188 @@
+
+    private static final String API_URL_CONFIG_PROP = "quarkus.log.handler.splunk.devservices.api-url";
+
+    private static final String SPLUNK_LATEST = "registry-1.docker.io/splunk/splunk:latest";
+
+    private static volatile DevServicesResultBuildItem.RunningDevService devService;
+
+    private static volatile DevServicesLoggingSplunkConfig capturedDevServiceConfig;
+
+    private static volatile boolean first = true;
+
+    /**
+     * Start one (or many in the future) SplunkContainer(s) depending on extension configuration. We also take care of
+     * locating and re-using existing container if configured in shared mode.
+     */
+    @BuildStep
+    public DevServicesResultBuildItem startSplunkContainer(LaunchModeBuildItem launchMode,
+            DockerStatusBuildItem dockerStatusBuildItem,
+            SplunkBuildConfig config,
+            Optional<ConsoleInstalledBuildItem> consoleInstalledBuildItem,
+            CuratedApplicationShutdownBuildItem closeBuildItem,
+            LoggingSetupBuildItem loggingSetupBuildItem,
+            GlobalDevServicesConfig devServicesConfig) {
+
+        // Figure out if we need to shut down and restart existing Splunk containers
+        // if not and the Splunk containers have already started we just return
+        if (devService != null) {
+            boolean restartRequired = !config.devservices.equals(capturedDevServiceConfig);
+            if (!restartRequired) {
+                return devService.toBuildItem();
+            }
+            try {
+                devService.close();
+            } catch (Throwable e) {
+                log.error("Failed to stop Splunk container", e);
+            }
+            devService = null;
+            capturedDevServiceConfig = null;
+        }
+
+        // Re-initialize captured config and dev services.
+        capturedDevServiceConfig = config.devservices;
+
+        StartupLogCompressor compressor = new StartupLogCompressor(
+                (launchMode.isTest() ? "(test) " : "") + "Microcks Dev Services Starting:", consoleInstalledBuildItem,
+                loggingSetupBuildItem);
+        DevServicesResultBuildItem.RunningDevService newDevService;
+        try {
+            newDevService = startContainer(config.devservices, dockerStatusBuildItem,
+                    launchMode.getLaunchMode(), devServicesConfig.timeout);
+
+            if (devService == null) {
+                compressor.closeAndDumpCaptured();
+            } else {
+                compressor.close();
+                log.infof("The Splunk container is ready on %s", devService.getConfig().get(HANDLER_URL_CONFIG_PROP));
+            }
+        } catch (Throwable t) {
+            compressor.closeAndDumpCaptured();
+            throw new RuntimeException(t);
+        }
+
+        // Save started Dev Services.
+        devService = newDevService;
+
+        if (devService == null) {
+            return null;
```

Comment:
This happen only if newDevService is null, we could return promptly if newDevService is null ?

Reply:
Removed newDevService intermediary variable